streams/src/main/java/org/apache/kafka/streams/kstream/internals/StatefulRepartitionNode.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+import org.apache.kafka.streams.processor.internals.InternalTopologyBuilder;
+import org.apache.kafka.streams.state.KeyValueStore;
+
+class StatefulRepartitionNode<K, V, T> extends RepartitionNode<K, V> {
+
+    private final ProcessorSupplier<K, Change<V>> statefulProcessorSupplier;
+    private final MaterializedInternal<K, T, KeyValueStore<Bytes, byte[]>> materialized;
+
+    StatefulRepartitionNode(final String predecessorNodeName,
+                            final String nodeName,
+                            final String sourceName,
+                            final Serde<K> keySerde,
+                            final Serde<V> valueSerde,
+                            final String sinkName,
+                            final String repartitionTopic,
+                            final String processorName,
+                            final ProcessorSupplier<K, Change<V>> statefulProcessorSupplier,
+                            final MaterializedInternal<K, T, KeyValueStore<Bytes, byte[]>> materialized) {
+        super(predecessorNodeName,
+              nodeName,
+              sourceName,
+              null,
+              keySerde,
+              valueSerde,
+              sinkName,
+              repartitionTopic,
+              processorName);
+
+        this.statefulProcessorSupplier = statefulProcessorSupplier;
+        this.materialized = materialized;
+    }
+
+    ProcessorSupplier<K, Change<V>> statefulProcessorSupplier() {
+        return statefulProcessorSupplier;
+    }
+
+    MaterializedInternal<K, T, KeyValueStore<Bytes, byte[]>> materialized() {
+        return materialized;
+    }
+
+    ChangedSerializer<? extends V> changedValueSerializer() {
```

Comment:
The name of this function is a bit weird (ditto below for value deserializer): are we expect that type `V` is `Change<V1>`? If no, why not just call `valueSerde().serializer()` directly?